moveit_simple/src/moveit_simple.cpp

```diff
@@ -220,8 +220,102 @@ bool Robot::getJointSolution(const Eigen::Affine3d &pose, double timeout,
 }
 
 
+
+bool Robot::getJointSolution(const Eigen::Affine3d &pose, const std::string& custom_tool_frame,
+                             double timeout, const std::vector<double> & seed,
+                             std::vector<double> & joint_point) const
+{
+  bool GET_JOINTS_FLAG = false;
+
+  std::string moveit_tool_link = joint_group_->getSolverInstance()->getTipFrame();
+
+  try
+  {
+      ROS_INFO_STREAM("Transforming Pose from custom_tool_frame frame [" << 
+                         moveit_tool_link << "] to moveit_end_link [" << 
+                         custom_tool_frame << "] before performing IK");
+
+      // Transform Target/Goal Point from custom tool frame to moveit_end_link
+      const Eigen::Affine3d custom_frame_goal_pose = 
+                               transformPoseBetweenFrames(pose, custom_tool_frame, 
+                                                          moveit_tool_link);
+      
+      std::lock_guard<std::recursive_mutex> guard(m_);
+
+      std::vector<double> local_seed = seed;
+      if (seed.empty())
+      {
+        ROS_INFO_STREAM("Empty seed passed to getJointSolution, using current state");
+        local_seed =  getJointState();
+      }
+
+      GET_JOINTS_FLAG = getIK(custom_frame_goal_pose, local_seed, joint_point, timeout);
+  }
+  catch(tf2::TransformException &ex)
+  {
+      ROS_WARN_STREAM("getJointSolution failed for arbitrary pose in Frame[" << 
```

Comment:
I know you don't have to explicitly set the flag since it is already initilaized above, but it helps readability if you set it to `false` in this block.

Reply:
Ok sure.